robot-server/robot_server/persistence/_migrations/up_to_3.py

```
@@ -109,25 +118,40 @@ def _migrate_run_table(
     insert_new_command = sqlalchemy.insert(schema_3.run_command_table)
 
     for old_run_row in source_transaction.execute(select_old_runs).all():
+        old_state_summary = old_run_row.state_summary
+        new_state_summary = (
+            None
+            if old_run_row.state_summary is None
+            else pydantic_to_json(
+                pydantic.parse_obj_as(StateSummary, old_state_summary)
+            )
+        )
```

We could add a `try: ... except: pass` around each run's migration to drop it in case something goes wrong, instead of making the whole server unavailable.

---

Yeah I've decided I should do this. Dropping runs silently is bad, but it's significantly better than failing the boot.

---

I haven't gotten to this yet. We should do it in a follow-up.